pkg/native/serializers/serializer_cdx.go

```diff
@@ -149,6 +155,13 @@ func (s *CDX) Serialize(bom *sbom.Document, _ *native.SerializeOptions, _ interf
 	return doc, nil
 }
 
+// isValidCycloneDXSerialNumber validates serial id against regex pattern
+func isValidCycloneDXSerialNumber(serial string) bool {
+	pattern := `^urn:uuid:[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$`
+	matched, _ := regexp.MatchString(pattern, serial)
```

Comment:
Define the regexp outside of the function and make sure we compile it only once when needed as dointg it at every function call is expensive.